File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn response-for [str]
  (cond (str/blank? str)          "Fine. Be that way!"
        (and (=    str (str/upper-case str))
             (not= str (str/lower-case str)))
                                  "Whoa, chill out!"
        (.endsWith str "?")       "Sure." ; #str/ends-with? is Clojure 1.8
        :else                     "Whatever."))

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.ends_with?(input, "?") ->
        "Sure."
      String.strip(input) == "" ->
        "Fine. Be that way!"
      String.upcase(input) == input && String.downcase(input) != input ->
        "Whoa, chill out!"
      true ->
        "Whatever."
    end
  end
end
